### ISSUE TITLE
Provide guidance as to how to fix non UTF-8 JSON response from "ansible.builtin.git" module

### ISSUE THREAD

**Issue by rayrutjes** [ansible/ansible#82820]
### Summary

I have the following deprecation warning:

```
[DEPRECATION WARNING]: Module "ansible.builtin.git" returned non UTF-8 data in
the JSON response. This will become an error in the future. This feature will
be removed in version 2.18. Deprecation warnings can be disabled by setting
deprecation_warnings=False in ansible.cfg.
```

It would be great to provide some guidance as to how to address this problem. I'm not sure how to workaround the problem, and whether there will be an option in the future to bypass the deprecation or not.

### Issue Type

Bug Report

### Component Name

git

### Additional Information

I would do 2 things.
1. Explicit whether there is a change needed because there will be no way of supporting non UTF-8 anymore, or whether there will an option, for example "unsafe" or whatever that would re-assure us to not be suddenly blocked by this deprecation
2. Provide some hints to what the error is related to and link to resources to fix the problem. 

I'm currently unsure exactly what needs to happen, whether it's related to the content of the git repo I'm cloning, or if it's something else. And in any case, not sure what we can or should do about it.

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

**Comment by ansibot** [comment]
Files identified in the description:

* [`lib/ansible/modules/git.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/git.py)

If these files are incorrect, please update the `component name` section of the description or use the [component](https://github.com/ansible/ansibotmini#commands) bot command.

**Comment by sivel** [comment]
Can you provide output from the `git` task causing this, with at least `-v`, or maybe `-vvv`?

I cannot see why the `git` module would have returned non-urf8 data.  If that is the case, it's a bug that needs addressed in the `git` module itself.

I suppose the only reason I can see is with `diff` where the contents of the git repo are not utf-8, in which case, the only possibility is a lossy operation in the git module to replace the non-UTF8 characters, to avoid the issue becoming an error, or not running with `diff`.

**Comment by sivel** [comment]
I have verified this warning, when a diff contains latin1 encoded characters.  As mentioned above, we have a few potential options:

1. Don't show a diff when it includes non-utf8 data
2. Perform a lossy operation to replace non-utf8 data with `?`

In both situations, we'd probably also need to display a warning as to why the diff isn't shown.

**Comment by sivel** [comment]
I've reclassified this as a bug, since it's not user resolvable, other than not using the diff option.

**Comment by seankene** [comment]
Just as an FYI -- I'm also getting a warning on this for GPG keys that have been provided to me from a third party software.  If this changes from a warning to an error, it will require installing that third party software a required manual process.

**Comment by jr-m4** [comment]
Slightly off-topic/module. But `community.crypto.get_certificate` also returns non UTF-8 data in the JSON response.

**Comment by rynstack** [comment]
> Just as an FYI -- I'm also getting a warning on this for GPG keys that have been provided to me from a third party software. If this changes from a warning to an error, it will require installing that third party software manually.

With ansible [core 2.19.4] I am also having this problem with GPG key download with the `shell` cmd module and `wget` piped into `gpg --dearmor` piped into `tee` output. For example:
`wget -O- https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB | gpg --dearmor | tee /etc/apt/keyrings/intel-oneapi_archive-keyring.gpg`

The task fails unless default config is adjusted.
``<<< caused by >>>
Refusing to deserialize an invalid UTF8 string value: 'utf-8' codec can't encode character '\udc99' in position 0: surrogates not allowed This check can be disabled with the `MODULE_STRICT_UTF8_RESPONSE` setting.``

So I switched to a new task module method instead
